src/Mod/Draft/drafttests/test_modification.py

```diff
@@ -592,6 +595,185 @@ def test_stretch(self):
+            ptNew = scale_vector_from_center(ends[i].add(base), sca, cen)
+            if not vrts[i].Point.isEqual(ptNew, 1e-8):
+                result = False
+                break
+        # check midpoints of arcs:
+        if result == True:
+            for i in range(4):
+                ptNew = scale_vector_from_center(mids[i].add(base), sca, cen)
+                edge = obj.Shape.Edges[i]
+                par = (edge.LastParameter - edge.FirstParameter) / 2
+                if not edge.valueAt(par).isEqual(ptNew, 1e-8):
+                    result = False
+                    break
+        self.assertTrue(result, "'{}' failed".format(operation))
+
+    def test_scale_part_feature_lines(self):
+        """Create and scale a part feature (lines)."""
+        operation = "Draft Scale part feature (lines)"
+        _msg("  Test '{}'".format(operation))
+
+        base = App.Vector(3.5, 2.5, 0)
+        cen = App.Vector(2, 1, 0) # center for scaling
+        sca = App.Vector(2, 3, 1)
+        pts = [App.Vector(0, 0, 0),
+               App.Vector(4, 0, 0),
+               App.Vector(4, 3, 0),
+               App.Vector(0, 3, 0)]
+
+        shp = Part.Shape([Part.LineSegment(pts[0], pts[1]),
+                          Part.LineSegment(pts[1], pts[2]),
+                          Part.LineSegment(pts[2], pts[3]),
+                          Part.LineSegment(pts[3], pts[0])])
+        obj = App.ActiveDocument.addObject("Part::Feature")
+        obj.Shape = shp
+        obj.Placement.Base = base
+        App.ActiveDocument.recompute()
+
+        _msg("  Scale")
+        _msg("  vector={}".format(sca))
+        _msg("  center={}".format(cen))
+        Draft.scale([obj], sca, cen, False)
+        App.ActiveDocument.recompute()
+
+        vrts = obj.Shape.Vertexes
+        result = True
+        for i in range(4):
+            ptNew = scale_vector_from_center(pts[i].add(base), sca, cen)
+            if not vrts[i].Point.isEqual(ptNew, 1e-8):
+                result = False
+                break
+        self.assertTrue(result, "'{}' failed".format(operation))
+
+    def test_scale_rectangle(self):
+        """Create and scale a rectangle."""
+        operation = "Draft Scale rectangle"
+        _msg("  Test '{}'".format(operation))
+
+        base = App.Vector(3.5, 2.5, 0)
+        cen = App.Vector(2, 1, 0) # center for scaling
+        sca = App.Vector(2, 3, 1)
+        len = 4
+        hgt = 3
+
+        obj = Draft.make_rectangle(len, hgt)
+        obj.Placement.Base = base
+        App.ActiveDocument.recompute()
+
+        _msg("  Scale")
+        _msg("  vector={}".format(sca))
+        _msg("  center={}".format(cen))
+        Draft.scale([obj], sca, cen, False)
+        App.ActiveDocument.recompute()
+
+        ptNew = scale_vector_from_center(base, sca, cen)
+        result = obj.Placement.Base.isEqual(ptNew, 1e-8)
+        if result == True:
+            result = math.isclose(obj.Length, len * sca.x, abs_tol = 1e-8)
```

Comment:
This fails in Python 2 because `math.isclose` doesn't exist in this version. If you want to compare for equality or closeness, I suggest using the methods of the `unittest` module itself. The class `unittest.TestCase` has all sorts of methods to compare values, `assertEqual`, `assertNotEqual`, `assertAlmostEqual`, etc.

[unittest.TestCase](https://docs.python.org/3/library/unittest.html#test-cases)

I think using these assertions is better than relying on other modules.

Reply:
I'll look into this.

Reply:
The unit tests in Travis still test for Python 2, so we still maintain Python 2 compatibility. Once we completely switch off those tests, then probably we can start using all sorts of Python 3 specific features.